.github/workflows/staging.yml

```diff
@@ -1,12 +1,52 @@
-name: Staging Deployment
+name: Deploy Staging
 
-on: workflow_dispatch
+on:
+  workflow_dispatch:
+    inputs:
+      deployMain:
+        description: "Deploy main? true/false"
+        required: false
+        default: "true"
 
 jobs:
-  terraform_plan:
-    name: Terraform plan
+  save_docker_image_tags_staging:
+    if: github.event.inputs.deployMain == 'true'
+    name: Copy and save image tags for staging environment
     runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.MCA_GITHUB_TOKEN }}
+      - name: Save image tags for staging environment
+        run: |
+          cd terraform
+          cp dev.images.tfvars staging.images.tfvars
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add .
+          git commit -m "Copy and save image tags for use in staging environment [skip ci]" & git push
 
+  terraform_deploy_staging:
+    if: always()
```

Comment:
Will also need to add a check that this is on the `main` branch, similar to how you have done the checks in the `main.yml` pipeline